networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1139 @@
+    >>> fwd_graph_entropy = [round(nx.graph_entropy(net, forward_entropy=True), 3) for net in condensed_network_layers]
+    >>> bkwd_graph_entropy = [round(nx.graph_entropy(net), 3) for net in condensed_network_layers]
+    >>> print("graph entropy (from top | bottom): {0} | {1}".format(fwd_graph_entropy, bkwd_graph_entropy))
+    graph entropy (from top | bottom): [0.52, 0.0] | [1.04, 0.0]
+
+    Notes
+    ______
+    WIP: Detail after confirmation of methodology from Prof. Sole.
+
+    As described in the text of _[1], proposed and developed in _[2], _[3]:
+    Graph Entropy, is measure of probability of crossing a node n_j when following a path from n_i.
+
+    _Forward_ entropy:
+    .. math:: H_f(G_C) = \frac{1}{|M|} \sum_{v_i \in M} \sum_{v_k \in V \setminus \mu} P(v_i \right v_k) \cdot log k_{out}(v_k)
+    _Backward_ entropy:
+    .. math:: H_b(G_C) = \frac{1}{|\mu|} \sum_{v_i \in \mu} \sum_{v_k \in V_C \setminus M} P(v_i \leftarrow v_k) \cdot log k_{in}(v_k)
+    where :math:`\mu` (M) is the set of minimal (maximal) nodes, and :math:`P(v_i \leftarrow v_k)` indicates
+    the probability of transition between :math:`v_k` and :math:`v_i`.
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    .. [2] "Topological reversibility and causality in feed-forward networks."
+      Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquín Goñi, and Ricard Solé.
+      New Journal of Physics 12, no. 11 (2010): pg 113051
+
+    .. [3] "Measuring the hierarchy of feedforward networks."
+    Corominas-Murtra, Bernat, Carlos Rodríguez-Caso, Joaquin Goni, and Ricard Solé.
+    Chaos: An Interdisciplinary Journal of Nonlinear Science 21, no. 1 (2011): pg 016108.
+
+    """
+    import numpy as np
+
+    if not nx.is_directed_acyclic_graph(DAG):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for graph entropy evaluation"
+        )
+
+    dag = nx.convert_node_labels_to_integers(DAG)
+    # Could be passed in most contexts to reduce redundant computation
+    L_GC = len(recursive_leaf_removal(DAG))
+
+    if forward_entropy:
+        B_prime = _matrix_normalize(nx.to_numpy_array(dag), axis=1)
+        # +1 as k \in ( 1, L(G_C) )
+        P = sum([np.power(B_prime, k) for k in range(1, L_GC + 1)])
+        # TODO: Not so sure about this unless k coincides with the number of steps already taken (and the sum is odd)
+        # (Presently awaiting response from original authors for clarification)
+    else:
+        # as w sklearn: B = normalize(nx.to_numpy_array(dag), axis=0, norm='max')  # Column normalization
+        B = _matrix_normalize(nx.to_numpy_array(dag), axis=0)
+        P = sum([np.power(B, k) for k in range(1, L_GC + 1)])
```

Comment:
I think this whole section collapses to a couple of lines now that axis is the only thing that changes.
The comments indicate that we maybe should be using `max` instead of `sum` to normalize. 
(i.e. use the infinity norm instead of the L_1 norm).   Thoughts?

If this does collapse, we should probably inline the _matrix_normalize

Reply:
Yes! I've collapsed this as suggested. The sklearn `max` norm would've been an improper implementation, as **B** is defined as the adjacency matrix normalized by total _in_ (backward entropy) or _out_ (forward entropy) degree. 
It appears `hierarchy_coordinates.py` does not use `_matrix_normalize` more than the once, though I'm not quite sure how to inline it. (Would seem to have too much state for a lambda function)